### PR TITLE
lib: nrf_modem: rename `lte_connectivity` -> `lte_net_if`

### PR DESCRIPTION
**Pull request by lemrey**
Splitting out the first few commits from the other PR.

### CODE REVIEW

File: lib/nrf_modem_lib/nrf91_sockets.c [hunk 1]
```diff
@@ -1158,13 +1158,13 @@ static void nrf91_iface_api_init(struct net_if *iface)
...
-#if defined(CONFIG_LTE_CONNECTIVITY)
-	return enabled ? lte_connectivity_enable() : lte_connectivity_disable();
+#if defined(CONFIG_NRF_MODEM_LIB_NET_IF)
+	return enabled ? lte_net_if_enable() : lte_net_if_disable();
 #else
 	ARG_UNUSED(iface);
 	ARG_UNUSED(enabled);
 	return 0;
-#endif /* CONFIG_NRF9160_CONNECTIVITY */
+#endif  /* CONFIG_NRF_MODEM_LIB_NET_IF */
```
Comment (eivindj-nordic) [inline lib/nrf_modem_lib/nrf91_sockets.c]: nit: extra space

**Review by eivindj-nordic — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Comment by lemrey** [comment]
@umapraseeda , I have added a line to the changelog, could you please have a look?

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -362,6 +362,7 @@ Modem libraries
 * :ref:`nrf_modem_lib_readme`:
 
    * Added a mention about enabling TF-M logging while using modem traces in the :ref:`modem_trace_module`.
+   * Renamed ``lte_connectivity`` module to ``lte_net_if`` and all related Kconfig options accordingly.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:365]:
```suggestion
* :ref:`nrf_modem_lib_readme`:

  * Added a mention about enabling TF-M logging while using modem traces in the :ref:`modem_trace_module`.
  * Updated by renaming ``lte_connectivity`` module to ``lte_net_if``.
    All related Kconfig options have been renamed accordingly.
```

File: include/modem/nrf_modem_lib.h [hunk 3]
```diff
@@ -196,6 +196,25 @@ struct nrf_modem_lib_shutdown_cb {
  */
 void nrf_modem_fault_handler(struct nrf_modem_fault_info *fault_info);
 
+#if defined(CONFIG_NRF_MODEM_LIB_NET_IF)
+/** @brief Options used to specify desired behavior when the network interface is brough down by
```
Comment (umapraseeda) [inline include/modem/nrf_modem_lib.h:200]:
```suggestion
/** @brief Options used to specify desired behavior when the network interface is brought down by
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (lemrey) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: THank you, updated.

File: include/modem/nrf_modem_lib.h (hunk 3 above)
Comment (lemrey) [reply include/modem/nrf_modem_lib.h]: Thank you, fixed.

**Review by umapraseeda — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
Looks great, thanks for separating these changes out

**Review by simensrostad — APPROVED** [review]
Looks good, thanks

**Pull request merged**